source/common/common/assert.h

```diff
@@ -322,5 +348,28 @@ void resetEnvoyBugCountersForTest();
 // but this macro replaces a less clear crash using NOT_REACHED_GCOVR_EXCL_LINE.
 #define PANIC_DUE_TO_CORRUPT_ENUM PANIC("corrupted enum");
 
+// CONDITION_STR is needed to prevent macros in condition from being expected, which obfuscates
+// the logged failure, e.g., "EAGAIN" vs "11".
+#define _ENVOY_NOTIFICATION_IMPL(CONDITION, NAME, DETAILS)                                         \
+  do {                                                                                             \
+    if (!(CONDITION)) {                                                                            \
```

Comment:
Everyone understands that `ASSERT` is supposed to have a true condtion. For this one, I think it's unclear when reading it when it would trigger. Wdyt of having it not take a condition at all, and usage being 

```
if (!cond) {
  ENVOY_NOTIFICATION("name", "some details");
}
```

and having the condition not be part of the macro?